### PR TITLE
file: Recognize Mach-O binaries and Java class files

### PR DESCRIPTION
**Pull request by mkhl**

### CODE REVIEW

File: src/cmd/file.c [hunk 1]
```diff
@@ -523,6 +523,9 @@ Filemagic long0tab[] = {
 	070707,		0xFFFF,		"cpio archive\n", OCTET,
 	0x2F7,		0xFFFF,		"tex dvi\n", "application/dvi\n",
 	0xfffa0000,	0xfffe0000,	"mp3 audio\n",	"audio/mpeg\n",
+	0xcafebabe,	0xFFFFFFFF,	"Mach-O executable\n",	"application/x-mach-binary\n",
+	0xfeedface,	0xFFFFFFFE,	"Mach-O executable\n",	"application/x-mach-binary\n",
+	0xbebafeca,	0xFFFFFFFF,	"Java class\n",	"application/x-java-applet\n",
```
Comment (kare) [inline src/cmd/file.c]: % hexdump -C -n 4 HelloWorld.class
00000000  ca fe ba be

http://radio-weblogs.com/0100490/2003/01/28.html

Comment (mkhl) [reply src/cmd/file.c]: Thanks for the link, such history is always interesting to read.
I’m not sure what your point is though?

~~~
% hexdump -C -n4 Main.class
00000000  ca fe ba be                                       |....|
00000004
% 9 file Main.class
Main.class: Java class
% hexdump -C -n4 Main
00000000  cf fa ed fe                                       |....|
00000004
% 9 file Main
Main: Mach-O executable
% uname -m -s
Darwin x86_64
~~~

Comment (kare) [reply src/cmd/file.c]: I'd expect 0xcafebabe to be filemagic for a Java class?

**Comment by mkhl** [comment]
I’m still not sure how to respond since these values work on my system. Have you tried the changeset?

Also, the magic constants are checked against `LENDIAN` of the buffer, which reverses the order of the first four bytes:

```
#define	LENDIAN(p)	((p)[0] | ((p)[1]<<8) | ((p)[2]<<16) | ((p)[3]<<24))
```

**Comment by rsc** [comment]
I was misreading the diff, sorry. Can you change the cafebabe entry to say "Mach-O fat executable" to make clear the difference between those two lines?

**Comment by rsc** [comment]
Merged.

**Pull request closed without merging**